Userland/Libraries/LibJS/Runtime/AsyncFromSyncIteratorPrototype.cpp

```diff
@@ -0,0 +1,163 @@
+    define_native_function(vm.names.return_, return_, 1, attr);
+    define_native_function(vm.names.throw_, throw_, 1, attr);
+}
+
+AsyncFromSyncIteratorPrototype::~AsyncFromSyncIteratorPrototype()
+{
+}
+
+// 27.1.4.4 AsyncFromSyncIteratorContinuation ( result, promiseCapability ), https://tc39.es/ecma262/#sec-asyncfromsynciteratorcontinuation
+static Object* async_from_sync_iterator_continuation(VM& vm, GlobalObject& global_object, Object& result, PromiseCapability& promise_capability)
+{
+    auto done = TRY_OR_REJECT(vm, promise_capability, iterator_complete(global_object, result));
+    auto value = TRY_OR_REJECT(vm, promise_capability, iterator_value(global_object, result));
+    auto value_wrapper = TRY_OR_REJECT(vm, promise_capability, promise_resolve(global_object, *global_object.promise_constructor(), value));
+
+    VERIFY(is<Promise>(value_wrapper));
+    auto* value_wrapper_promise = static_cast<Promise*>(value_wrapper);
+
+    value_wrapper_promise->perform_then(
+        NativeFunction::create(global_object, "", [done](VM& vm, GlobalObject& global_object1) -> ThrowCompletionOr<Value> {
+            return create_iterator_result_object(global_object1, vm.argument(0), done);
+        }),
+        js_undefined(), promise_capability);
+    return promise_capability.promise;
+}
+
+// 27.1.4.2.1 %AsyncFromSyncIteratorPrototype%.next ( [ value ] ), https://tc39.es/ecma262/#sec-%asyncfromsynciteratorprototype%.next
+JS_DEFINE_NATIVE_FUNCTION(AsyncFromSyncIteratorPrototype::next)
+{
+    auto* this_object = TRY(typed_this_object(global_object));
+    auto promise_capability = TRY(new_promise_capability(global_object, global_object.promise_constructor()));
+
+    ThrowCompletionOr<Object*> result(nullptr);
+    if (vm.argument_count() > 0)
+        result = iterator_next(this_object->sync_iterator_record(), vm.argument(0));
+    else
+        result = iterator_next(this_object->sync_iterator_record());
+
+    // Since result comes from two expressions we do IfAbruptRejectPromise explicitly here.
+    if (result.is_throw_completion()) {
+        vm.clear_exception();
+        vm.stop_unwind();
+        (void)vm.call(*promise_capability.reject, js_undefined(), result.release_error().value());
+        return promise_capability.promise;
+    }
+
+    return async_from_sync_iterator_continuation(vm, global_object, *result.value(), promise_capability);
+}
+
+// 27.1.4.2.2 %AsyncFromSyncIteratorPrototype%.return ( [ value ] ), https://tc39.es/ecma262/#sec-%asyncfromsynciteratorprototype%.return
+JS_DEFINE_NATIVE_FUNCTION(AsyncFromSyncIteratorPrototype::return_)
+{
+    auto* this_object = TRY(typed_this_object(global_object));
+    auto promise_capability = TRY(new_promise_capability(global_object, global_object.promise_constructor()));
+    auto& sync_iterator = this_object->sync_iterator_record();
+    auto* return_method = TRY_OR_REJECT(vm, promise_capability, sync_iterator.value_of().get_method(global_object, vm.names.return_));
```

Comment:
Please explicitly wrap objects in a `Value()` call instead of using their `value_of()` method.